plugins/plugins-api/src/main/java/org/codice/compliance/saml/plugin/IdpRedirectResponse.java

```diff
@@ -78,68 +71,19 @@ protected IdpRedirectResponse(IdpRedirectResponse response) {
     super(response);
     url = response.url;
     path = response.path;
-    samlEncoding = response.samlEncoding;
-    sigAlg = response.sigAlg;
-    signature = response.signature;
+    parameters = response.parameters;
   }
 
   // TODO remove url field if not used in Binding Verification
-  private String url;
-  private String path;
-  private String samlEncoding;
-  private String sigAlg;
-  private String signature;
-
-  private boolean extraUrlParameters;
+  protected String url;
+  protected String path;
+  protected String parameters;
 
   @SuppressWarnings("squid:S3398" /* Method in here to simplify builder class */)
   private void parseAndSetUrlValues(String url) {
     this.url = url;
-
-    String urlPath = url.split("\\?")[0];
-    String urlParameters = url.split("\\?")[1];
-
-    path = urlPath;
-    extraUrlParameters = false;
-
-    for (String parameter : Splitter.on("&").split(urlParameters)) {
-      if (parameter.startsWith(SAML_RESPONSE + "=")) {
-        samlResponse = parameter.replace(SAML_RESPONSE + "=", "");
-      } else if (parameter.startsWith(SIG_ALG + "=")) {
-        sigAlg = parameter.replace(SIG_ALG + "=", "");
-      } else if (parameter.startsWith(SIGNATURE + "=")) {
-        signature = parameter.replace(SIGNATURE + "=", "");
-      } else if (parameter.startsWith(RELAY_STATE + "=")) {
-        relayState = parameter.replace(RELAY_STATE + "=", "");
-      } else if (parameter.startsWith("SAMLEncoding=")) {
-        samlEncoding = parameter.replace("SAMLEncoding=", "");
-      } else {
-        extraUrlParameters = true;
-      }
-    }
-  }
-
-  public String getUrl() {
-    return url;
-  }
-
-  public String getPath() {
-    return path;
-  }
-
-  public String getSigAlg() {
-    return sigAlg;
-  }
-
-  public String getSignature() {
-    return signature;
-  }
-
-  public String getSamlEncoding() {
-    return samlEncoding;
-  }
-
-  public boolean isExtraUrlParameters() {
-    return extraUrlParameters;
+    String[] splitUrl = url.split("\\?");
```

Comment:
❓ Have you run a build since rebasing? This would have failed errorprone, I believe.